relational_operators/HashJoinOperator.hpp

```diff
@@ -205,16 +256,23 @@ class HashJoinOperator : public RelationalOperator {
   const bool any_join_key_attributes_nullable_;
   const CatalogRelation &output_relation_;
   const QueryContext::insert_destination_id output_destination_index_;
-  const QueryContext::join_hash_table_id hash_table_index_;
+  std::vector<QueryContext::join_hash_table_id> hash_table_index_;
```

Comment:
I believe we could still mark `const` here. Also, please rename to `hash_table_indexes_` or `hash_table_indices_`. Thanks!